build.sh

```diff
@@ -0,0 +1,20 @@
+CXXTEST_VERSION=4.4
```

Comment:
Why don't you do that directly in a Dockerfile?

Reply:
Agreed. The Docker file should be self-contained so that it can built without any external help.

This is probably a great opportunity to try the new [multi-stage builds](https://docs.docker.com/engine/userguide/eng-image/multistage-build/#use-multi-stage-builds) feature.

Reply:
Knowing that I use the version of CxxTest in the `build.sh` to tag the image, should I pass it to the Dockerfile as an argument, or should I just copy the value in both files?

Reply:
I would recommend using git tag information to tag the image, because a lot of docker build services support it (hub.docker.com does, for example). Since it's a bit hard to automatically go from Dockerfile to git tag or vice versa, I'd just set the tag manually.

I think the right way to think if it is to *not* have a build.sh at all.

Reply:
OK, I've removed the build.sh and I've used a multi-stage build.
The version number is now only in the Dockerfile.

@jbruggem what do we need to be ble to publish the image on docker hub? Is it OK that the Dockerfile is in the _docker_ subfolder?

Reply:
I think it's ok, but I don't see the point of keeping it in that sub-folder, TBH :)

Reply:
Well, the [best practices](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#use-a-dockerignore-file) say that Dockerfiles should be in their own folder so that you can control the context of the docker build, but in this case we might not really care because this repository will always be very small.

What should I do? What do you want me to do?

Reply:
The decision is yours :)

Reply:
I think the mentioned best practice is for a project not a lonely Docker image. Except when the build of the image is complex, the Dockerfile is usually put at the root when the repository is only about the Docker Image.

Reply:
OK, I've moved the Dockerfile to the root.